packages/cli/src/utils/preprocess.ts

```diff
@@ -55,6 +55,26 @@ export class DatabaseTaskPreProcess {
           },
         }
       : { instanceId: '8f97ca29-4259-4fe7-a441-29e04f5edaf9' } // Current deployed version of service database
+
+    if (step.taskKey === 'list') {
+      const conditions = Array.isArray(step.inputs.conditions)
+        ? step.inputs.conditions.map((condition: any) => ({
```

Comment:
This will fail with no conditions.

```suggestion
        ? (step.inputs.conditions || []).map((condition: any) => ({
```